### PR TITLE
feat!: update to new Wallet and DPNS contract

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Updating `DPNS` methods to use new `DPNS` contract

## What was done?

- Updated `register` and `resolve` methods

## How Has This Been Tested?

Unit tests.

## Breaking Changes

`client.platform.names.register` now receive records as a second argument.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by jawid-h** [comment]
> We need to update docs + method definition (as we use records instead of names for register.names).
> Doing so right now (cc: @jawid-h )

@Alex-Werner Good catch, my bad!

File: src/SDK/Client/Platform/methods/names/register.ts [hunk 1]
```diff
@@ -35,25 +38,21 @@ export async function register(this: Platform,
     const [label] = nameLabels;
     const normalizedLabel = label.toLowerCase();
 
-    const preorderSalt = entropy.generate();
+    const preorderSalt = bs58.decode(entropy.generate());
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/names/register.ts:41]: This won't work. Contract requires just random 32 bytes. Let's use `crypto.randomBytes(32)`;

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
   "dependencies": {
     "@dashevo/dapi-client": "~0.15.0-dev.1",
```
Comment (shumkov) [inline package.json:34]: Let's update other stuff too?

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — DISMISSED** [review]
LGTM 👍

**Review by Alex-Werner — APPROVED** [review]
LGTM!

**Review by shumkov — APPROVED** [review]
LGTM 👍

**Pull request merged**